### ISSUE TITLE
Conflict resolution for replaced arrays and objects

### ISSUE THREAD

**Issue by mpareja** [automerge/automerge#530]
TL;DR: Array and object values returned by `Automerge.getConflicts` cannot be assigned to a document as part of conflict resolution.

---

The [documentation for managing conflicts](https://automerge.org/docs/cookbook/conflicts/) does not demonstrate how to use the conflict information to resolve a conflict, so I could be doing this wrong. I assumed I would be able to use the values in the conflict information directly to resolve a conflict like so:

```js
const result = Automerge.merge(doc1, doc2)
const conflicts = Automerge.getConflicts(result, 'list')

Automerge.change(doc2, d => {
  d.list = Object.values(conflicts).pop(); // bogus "pop" resolution strategy :)
});
```

While this works for primitive values, the approach yields a `RangeError: Cannot create a reference to an existing document object` for arrays and objects. The value portion of the `opId -> value` map includes `_objectId` and other Automerge symbols.

Am I using `Automerge.getConflicts` incorrectly or should it be returning a subtly different result?

#### Test Case
```js
let doc1 = Automerge.init()
doc1 = Automerge.change(doc1, d => d.list = [])
let doc2 = Automerge.clone(doc1)

doc1 = Automerge.change(doc1, d => d.list = [])
doc2 = Automerge.change(doc2, d => d.list = [])

const result = Automerge.merge(doc1, doc2)
const conflicts = Automerge.getConflicts(result, 'list')

// the following throws: RangeError: Cannot create a reference to an existing document object
Automerge.change(doc2, d => d.list = Object.values(conflicts)[0])
```
